plugin/src/main/java/org/opensearch/ml/utils/SecurityUtils.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.ml.utils;
+
+import static org.opensearch.ml.common.CommonValue.ML_MODEL_GROUP_INDEX;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import lombok.extern.log4j.Log4j2;
+
+import org.apache.commons.lang3.StringUtils;
+import org.opensearch.action.ActionListener;
+import org.opensearch.action.get.GetRequest;
+import org.opensearch.client.Client;
+import org.opensearch.common.util.CollectionUtils;
+import org.opensearch.commons.authuser.User;
+import org.opensearch.index.query.BoolQueryBuilder;
+import org.opensearch.index.query.QueryBuilder;
+import org.opensearch.index.query.QueryBuilders;
+import org.opensearch.index.query.TermsQueryBuilder;
+import org.opensearch.ml.common.MLModelGroup;
+import org.opensearch.ml.common.exception.MLValidationException;
+import org.opensearch.search.builder.SearchSourceBuilder;
+
+import com.google.common.collect.ImmutableList;
+
+@Log4j2
+public class SecurityUtils {
+
+    public static void validateModelGroupAccess(User user, String modelGroupId, Client client, ActionListener<Boolean> listener) {
+        if (modelGroupId == null || isAdmin(user)) {
+            listener.onResponse(Boolean.TRUE);
```

Comment:
Is there any specific reason why using encapsulated boolean type here?

Reply:
No specific reason. We can use the primitive ones here.